gapic-generator/lib/gapic/presenters/method/rest_pagination_info.rb

```diff
@@ -0,0 +1,180 @@
+# frozen_string_literal: true
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Gapic
+  module Presenters
+    module Method
+      ##
+      # Pagination info determined from the proto method
+      #
+      class RestPaginationInfo
+        ##
+        # @param proto_method [Gapic::Schema::Method] the method to derive pagination info from
+        #
+        def initialize proto_method
+          @request = proto_method.input
+          @response = proto_method.output
+          @server_streaming = proto_method.server_streaming
+        end
+
+        ##
+        # Whether the method should be generated as paged
+        #
+        # @return [Boolean]
+        def paged?
+          !server_streaming? && paged_request? && paged_response?
+        end
+
+        ##
+        # Name of the request's field used for page size
+        # For Regapic can be either `page_size` or `max_results`
+        #
+        # @return [String, Nil]
```

Comment:
Nit: use `nil` (lower-case "n"). There is no `Nil` class name.

Reply:
:+1: